docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,112 @@
+# Syncing LDAP Groups to OpenShift Groups
+
+OpenShift cluster admins need a way to sync their OpenShift `Groups` to external records in an LDAP server. This requires that we have a way of retrieving the LDAP user entries that are the members of a group, determining the OpenShift `Group` name for this collection, and updating the OpenShift record, removing the record as necessary when the corresponding LDAP record is removed. Necessary information about the LDAP server set-up and schema will be provided by the user in a `yaml` configuration file and pointed to in the `oadm sync-groups` command.
+
+### Use Cases
+1. I want to sync all LDAP groups minus some blacklist to OpenShift `Groups`
+* I want to sync a specific set of LDAP groups to OpenShift `Groups`
+* I want to sync all OpenShift groups minus some blacklist with the LDAP server
+* I want to sync specific OpenShift `Groups` with the LDAP server records
+* I want to remove OpenShift `Groups` that were previously backed by LDAP records that no longer exist
+* I want to specify an explicit mapping from LDAP group identifiers to OpenShift `Group` resources
+
+## Supported LDAP Group Schemas
+The three LDAP schemas supported are
+
+1. groups as first-class LDAP entries that contain lists of members of the group
+2. users as first-class LDAP entries that contain lists of groups they are a member of
+3. schema 2 above, also including first-class groups to store metadata
+
+### Required OpenShift `Group` Metadata
+OpenShift `Group`'s `Annotations` will be used to store metadata regarding the sync process. `Annotations` will store the LDAP unique identifier (`openshift.io/ldap-uid`), the last sync time (`openshift.io/ldap-last-sync-time`), and, if available, the LDAP resource version (`openshift.io/ldap-resource-version`). A `Label` will also be set to determine the external record provider (`openshift.io/external-record-provider=LDAP`).
```

Comment:
Check to see if the extended label selection made it in.  If so, we should be able to select on the existence of a label.  That would allow us to to store the ldap server url instead of the value LDAP.  That would make precise filtering easier.

If not, make an annotation for ldap URL